src/ace/ari_text/parsemod.py

```diff
@@ -326,8 +339,9 @@ def p_structlist_end(p):
 
 
 def p_structpair(p):
-    'structpair : VALSEG EQ VALSEG SC'
-    # Keys are case-insensitive so get folded to lower case
+    '''structpair : VALSEG EQ VALSEG SC 
+                  | VALSEG EQ typedlit SC'''
```

Comment:
Change looks great! Only suggestion here is to keep the old comment about case folding.